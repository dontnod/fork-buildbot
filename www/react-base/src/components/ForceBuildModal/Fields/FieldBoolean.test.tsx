/*
  This file is part of Buildbot.  Buildbot is free software: you can
  redistribute it and/or modify it under the terms of the GNU General Public
  License as published by the Free Software Foundation, version 2.

  This program is distributed in the hope that it will be useful, but WITHOUT
  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
  details.

  You should have received a copy of the GNU General Public License along with
  this program; if not, write to the Free Software Foundation, Inc., 51
  Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

  Copyright Buildbot Team Members
*/

import {describe, expect, it} from "vitest";
import renderer from 'react-test-renderer';
import { FieldBoolean } from "./FieldBoolean";
import { ForceSchedulerFieldBoolean } from 'buildbot-data-js';
import { ForceBuildModalFieldsState } from '../ForceBuildModalFieldsState';

function assertRenderToSnapshot(defaultValue: boolean, stateValue?: boolean, updateValue: boolean = false) {
  const field: ForceSchedulerFieldBoolean = {
    name: 'dummy',
    fullName: 'fullDummy',
    label: 'dummyLabel',
    tablabel: 'dummyTabLabel',
    type: 'bool',
    default: defaultValue,
    multiple: false,
    regex: null,
    hide: false,
    maxsize: null,
    autopopulate: null,
  }
  const state = new ForceBuildModalFieldsState();
  state.createNewField(field.fullName, field.default);
  if (stateValue !== undefined) {
    state.setValue(field.fullName, stateValue);
  }

  const component = renderer.create(
    <FieldBoolean field={field} fieldsState={state} />
  );
  expect(component.toJSON()).toMatchSnapshot();

  if (updateValue) {
    const previousState = state.getValue(field.fullName);
    const expectedState = !previousState;
    renderer.act(() => {
      const elements = component.root.findAllByProps({'data-bb-test-id': `force-field-${field.fullName}`}, {deep: true});
      expect(elements.length).toBe(1);
      const checkbox = elements[0];
      checkbox.props.onChange({target: {checked: expectedState}});
    });
    expect(state.getValue(field.fullName)).toBe(expectedState);
  }
}

describe('ForceFieldBoolean component', function () {
  it('render default value False', () => {
    assertRenderToSnapshot(false);
  });

  it('render default value True', () => {
    assertRenderToSnapshot(true);
  });

  it('render non-default value False', () => {
    assertRenderToSnapshot(true, false);
  });

  it('render non-default value True', () => {
    assertRenderToSnapshot(false, true);
  });

  it('change state on click', () => {
    assertRenderToSnapshot(true, true, true);
  });
});
