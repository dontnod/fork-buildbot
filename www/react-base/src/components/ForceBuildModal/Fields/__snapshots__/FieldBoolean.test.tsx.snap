// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForceFieldBoolean component > change state on click 1`] = `
<div
  className="form-group"
>
  <div
    className="col-sm-10 col-sm-offset-2"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          data-bb-test-id="force-field-fullDummy"
          onChange={[Function]}
          type="checkbox"
        />
         
        dummyLabel
      </label>
    </div>
  </div>
</div>
`;

exports[`ForceFieldBoolean component > render default value False 1`] = `
<div
  className="form-group"
>
  <div
    className="col-sm-10 col-sm-offset-2"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          data-bb-test-id="force-field-fullDummy"
          onChange={[Function]}
          type="checkbox"
        />
         
        dummyLabel
      </label>
    </div>
  </div>
</div>
`;

exports[`ForceFieldBoolean component > render default value True 1`] = `
<div
  className="form-group"
>
  <div
    className="col-sm-10 col-sm-offset-2"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          data-bb-test-id="force-field-fullDummy"
          onChange={[Function]}
          type="checkbox"
        />
         
        dummyLabel
      </label>
    </div>
  </div>
</div>
`;

exports[`ForceFieldBoolean component > render non-default value False 1`] = `
<div
  className="form-group"
>
  <div
    className="col-sm-10 col-sm-offset-2"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={false}
          data-bb-test-id="force-field-fullDummy"
          onChange={[Function]}
          type="checkbox"
        />
         
        dummyLabel
      </label>
    </div>
  </div>
</div>
`;

exports[`ForceFieldBoolean component > render non-default value True 1`] = `
<div
  className="form-group"
>
  <div
    className="col-sm-10 col-sm-offset-2"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          checked={true}
          data-bb-test-id="force-field-fullDummy"
          onChange={[Function]}
          type="checkbox"
        />
         
        dummyLabel
      </label>
    </div>
  </div>
</div>
`;
